node/nodeRunner.go

```diff
@@ -169,16 +169,29 @@ func (nr *nodeRunner) startShufflingProcessLoop(
 			return err
 		}
 
+		nr.shuffleOutStatsAndGC()
+	}
+}
+
+func (nr *nodeRunner) shuffleOutStatsAndGC() {
+	debugConfig := nr.configs.GeneralConfig.Debug.ShuffleOut
+
+	if debugConfig.ExtraPrintsOnShuffleOut {
 		log.Debug("node statistics before running GC", statistics.GetRuntimeStatistics()...)
+	}
+	if debugConfig.CallGCWhenShuffleOut {
 		runtime.GC()
-		log.Debug("node statistics after running GC", statistics.GetRuntimeStatistics()...)
-
-		parentPath := filepath.Join(nr.configs.FlagsConfig.WorkingDir, nr.configs.GeneralConfig.Health.FolderPath)
-		var stats runtime.MemStats
-		runtime.ReadMemStats(&stats)
-		err = health.WriteMemoryUseInfo(stats, time.Now(), parentPath, "softrestart")
-		log.LogIfError(err)
 	}
+	if !debugConfig.ExtraPrintsOnShuffleOut {
+		return
+	}
+	log.Debug("node statistics after running GC", statistics.GetRuntimeStatistics()...)
```

Comment:
this might not be "after running GC" if configured to false in config

Reply:
fixed